webviz_subsurface/private_containers/_surface_selector.py

```diff
@@ -0,0 +1,586 @@
+                return [], None, self.hide_dropdown_style
+            value = current_value if current_value in sens_names else sens_names[0]
+            options = [{"value": sens, "label": sens} for sens in sens_names]
+            return options, value, self.show_dropdown_style
+
+        @app.callback(
+            [
+                Output(self.sens_case_id, "options"),
+                Output(self.sens_case_id, "value"),
+                Output(self.sens_case_wrapper_id, "style"),
+            ],
+            [Input(self.sens_name_id, "value")],
+            [State(self.ensemble_id, "value"), State(self.sens_case_id, "value")],
+        )
+        def _update_sens_case(sensname, ensemble, current_value):
+            sens_cases = self.sens_cases(ensemble, sensname)
+            if not sens_cases:
+                return [], None, self.hide_dropdown_style
+            value = current_value if current_value in sens_cases else sens_cases[0]
+            options = [{"value": sens, "label": sens} for sens in sens_cases]
+            return options, value, self.show_dropdown_style
+
+        @app.callback(
+            Output(self.storage_id, "children"),
+            [
+                Input(self.attr_id, "value"),
+                Input(self.name_id, "value"),
+                Input(self.date_id, "value"),
+                Input(self.ensemble_id, "value"),
+                Input(self.aggreal_id, "value"),
+                Input(self.realization_id, "value"),
+                Input(self.sens_name_id, "value"),
+                Input(self.sens_case_id, "value"),
+            ],
+        )
+        def _set_data(
+            attr, name, date, ensemble, aggreal, calculation, sens_name, sens_case
+        ):
+
+            """
+            Stores current selections to dcc.Store. The information can be retrieved as a json string
+            from a dash callback Input. E.g. [Input(surfselector.storage_id, 'children')]
+            """
+            reals = self.realizations(ensemble, sens_name, sens_case)
+            all_senscases = [
+                {
+                    "case": senscase,
+                    "realizations": self.realizations(ensemble, sens_name, senscase),
+                }
+                for senscase in self.sens_cases(ensemble, sens_name)
+            ]
+            return json.dumps(
+                {
+                    "attribute": attr,
+                    "name": name,
+                    "date": date,
+                    "ensemble": ensemble,
+                    "aggregation": calculation if aggreal == "Aggregation" else None,
+                    "realization": reals if aggreal == "Aggregation" else calculation,
+                    "sensname": sens_name,
+                    "senscase": sens_case,
+                    "all_senscases": all_senscases,
+                }
+            )
+
+
+def prev_value(current_value, options):
+    try:
+        index = options.index(current_value)
+    except ValueError:
```

Comment:
In which scenario does this happen (i.e. that the current value is not among the options)?

Reply:
The options in one dropdown are updated based on selections in another dropdown.
The callback are handling both setting new options and setting the current value from existing options or new options. 
When this function is called it could be that the options have changed.